src/connector/src/parser/additional_columns.rs

```diff
@@ -281,9 +281,15 @@ pub fn build_additional_column_desc(
 pub fn source_add_partition_offset_cols(
     columns: &[ColumnCatalog],
     connector_name: &str,
+    skip_col_id: bool,
 ) -> ([bool; 2], [ColumnDesc; 2]) {
     let mut columns_exist = [false; 2];
-    let mut last_column_id = max_column_id(columns);
+    let mut last_column_id = if skip_col_id {
+        // col id will be filled outside later. Here just use a placeholder.
+        ColumnId::new(0)
+    } else {
+        max_column_id(columns)
+    };
```

Comment:
I don't see where we fill the `ColumnId`, can you elaborate?

Reply:
Is this really safe?

Reply:
https://github.com/risingwavelabs/risingwave/blob/39e77fbfa11f446cb5716fe677f74adb72916c26/src/frontend/src/handler/create_source.rs#L1600-L1605

Reply:
😄🤔 What is `chenge`?

Reply:
Can you link to the function in `create_source.rs` from the comment of `ColumnId::new(0)`? I don't feel confident in terms of future maintanance.

Reply:
+1, why we need a placeholder here

Reply:
please use `ColumnId::placeholder()`

Reply:
> Because at this stage the col ids are `ColumnId::placeholder()` (`i32::MAX - 1`), and will overflow here.

Then I suppose doing incrementation on a dummy column ID can be worse.🤔

Reply:
updated